File: scala-3.6/template-renderer/src/main/scala/com/rbleggi/templaterenderer/core/RendererFactory.scala
package com.rbleggi.templaterenderer.core

import com.rbleggi.templaterenderer.renderers.{CSVRenderer, HTMLRenderer, PDFRenderer}

object RendererFactory {
  def getRenderer(format: String): TemplateRenderer = format.toLowerCase match
    case "html" => new HTMLRenderer()
    case "csv" => new CSVRenderer()
    case "pdf" => new PDFRenderer()
    case _ => throw new IllegalArgumentException(s"Unknown format: $format")
}

File: kotlin/template-renderer/src/main/kotlin/com/rbleggi/templaterenderer/service/RendererFactory.kt
package com.rbleggi.templaterenderer.service

import com.rbleggi.templaterenderer.service.renderer.CsvRenderer
import com.rbleggi.templaterenderer.service.renderer.HtmlRenderer
import com.rbleggi.templaterenderer.service.renderer.PdfRenderer
import com.rbleggi.templaterenderer.service.renderer.TemplateRenderer

object RendererFactory {
    fun getRenderer(type: String): TemplateRenderer = when (type.lowercase()) {
        "html" -> HtmlRenderer()
        "csv" -> CsvRenderer()
        "pdf" -> PdfRenderer()
        else -> throw IllegalArgumentException("Unknown format: $type")
    }
}
